### PR TITLE
chore(config): RHICOMPL-2561 Enabling has_many inversing

### PR DESCRIPTION
**Pull request by romanblanco**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/models/concerns/profile_policy_association.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def destroy_policy_with_internal
     end
 
     def destroy_empty_policy
-      return unless policy&.profiles&.empty?
+      return if !(policy&.profiles&.empty? && policy&.profiles&.all? { |p| p.destroyed? })
```
Comment (skateman) [inline app/models/concerns/profile_policy_association.rb:55]: `any?` and `none?` maybe?

Comment (romanblanco) [reply app/models/concerns/profile_policy_association.rb:55]: @skateman
To clarify, is the idea to proceed if there is _at least one_ destroyed profile?
Shouldn't we only proceed if _all_ profiles are destroyed?

File: app/models/concerns/profile_policy_association.rb [hunk 2]
```diff
@@ -52,7 +52,9 @@ def destroy_policy_with_internal
     end
 
     def destroy_empty_policy
-      return unless policy&.profiles&.empty?
+      return if policy&.profiles&.reload&.any? && !policy&.profiles&.all? do |p|
+        p.canonical? || p.destroyed?
+      end
```
Comment (skateman) [inline app/models/concerns/profile_policy_association.rb:57]: Do you really need this after the reload?

File: config/application.rb [hunk 3]
```diff
@@ -52,8 +52,7 @@ class Application < Rails::Application
     require 'satellite_compensation/middleware'
     config.middleware.use Insights::API::Common::SatelliteCompensation::Middleware
 
-    # FIXME: profile<->policy connections could fail on this
-    config.active_record.has_many_inversing = false
+    config.active_record.has_many_inversing = true
```
Comment (skateman) [inline config/application.rb:55]: The `true` value is the default, so you can just drop this line.

File: app/models/concerns/profile_policy_association.rb (hunk 1 above)
Comment (skateman) [reply app/models/concerns/profile_policy_association.rb]: We're not changing any feature in this PR, my hint was only to reduce the complexity of the conditions you created.

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**